spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ControllerHelperServiceUtils.java

```diff
@@ -1,19 +1,125 @@
 package uk.ac.bbsrc.tgac.miso.spring;
 
+import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.isStringEmptyOrNull;
+
 import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 import javax.servlet.http.HttpSession;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import com.eaglegenomics.simlims.core.User;
+import com.eaglegenomics.simlims.core.manager.SecurityManager;
+
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import net.sourceforge.fluxion.ajax.util.JSONUtils;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Barcodable;
+import uk.ac.bbsrc.tgac.miso.core.data.Printer;
+import uk.ac.bbsrc.tgac.miso.service.PrinterService;
+
 /**
  * For when you need web utils, since this module can't access `miso-web`.
  *
  */
 public class ControllerHelperServiceUtils {
+  private static final Logger log = LoggerFactory.getLogger(ControllerHelperServiceUtils.class);
+
+  public interface BarcodePrintAssister<T extends Barcodable> {
+
+    public Iterable<T> fetchAll(long projectId) throws IOException;
+
+    public T fetch(long id) throws IOException;
+
+    public void store(T item, User user) throws IOException;
```

Comment:
It seems the User parameter is only for updating lastModifier? If so, remove. This should be done at service/requestman level